R/step_date_hcai.R

```diff
@@ -0,0 +1,269 @@
+#' available for features `month` or `dow`.
+#' @param columns A character string of variables that will be
+#' used as inputs. This field is a placeholder and will be
+#' populated once [prep.recipe()] is used.
+#' @param trained A logical to indicate if the number of NA values have been
+#' counted in preprocessing.
+#' @param skip A logical. Should the step be skipped when the
+#' recipe is baked?
+#' @return For `step_date_hcai`, an updated version of recipe with
+#' the new step added to the sequence of existing steps (if any).
+#' For the `tidy` method, a tibble with columns `terms`
+#' (the selectors or variables selected), `value` (the feature
+#' names), and `ordinal` (a logical).
+#' @export
+#' @details Unlike other steps, `step_date_hcai` does *not*
+#' remove the original date variables. [step_rm()] can be
+#' used for this purpose.
+#' @examples
+#' library(lubridate)
+#' library(recipes)
+#'
+#' examples <- data.frame(Dan = ymd("2002-03-04") + days(1:10),
+#'                        Stefan = ymd("2006-01-13") + days(1:10))
+#' date_rec <- recipe(~ Dan + Stefan, examples) %>%
+#'   step_date_hcai(all_predictors())
+#'
+#' date_rec <- prep(date_rec, training = examples)
+#'
+#' date_values <- bake(date_rec, newdata = examples)
+#' date_values
+#'
+step_date_hcai <-
+  function(recipe,
+           ...,
+           role = "predictor",
+           trained = FALSE,
+           features = c("dow", "month", "year"),
+           abbr = TRUE,
+           label = TRUE,
+           ordinal = FALSE,
+           columns = NULL,
+           skip = FALSE
+  ) {
+    feat <-
+      c("year",
+        "doy",
+        "week",
+        "decimal",
+        "semester",
+        "quarter",
+        "dow",
+        "month")
+    if (!all(features %in% feat))
+      stop("Possible values of `features` should include: ",
+           paste0("'", feat, "'", collapse = ", "))
+    add_step(
+      recipe,
+      step_date_hcai_new(
+        terms = ellipse_check(...),
+        role = role,
+        trained = trained,
+        features = features,
+        abbr = abbr,
+        label = label,
+        ordinal = ordinal,
+        columns = columns,
+        skip = skip
+      )
+    )
+  }
+
+step_date_hcai_new <-
+  function(
+    terms = NULL,
+    role = "predictor",
+    trained = FALSE,
+    features = features,
+    abbr = abbr,
+    label = label,
+    ordinal = ordinal,
+    columns = columns,
+    skip = FALSE
+  ) {
+    step(
+      subclass = "date_hcai",
+      terms = terms,
+      role = role,
+      trained = trained,
+      features = features,
+      abbr = abbr,
+      label = label,
+      ordinal = ordinal,
+      columns = columns,
+      skip = skip
+    )
+  }
+
+#' @importFrom stats as.formula model.frame
+#' @export
+prep.step_date_hcai <- function(x, training, info = NULL, ...) {
+  col_names <- terms_select(x$terms, info = info)
+  date_data <- info[info$variable %in% col_names, ]
+  # if (any(date_data$type != "date"))
```

Comment:
Not sure how these escaped lintr, but commented lines should be removed (if the test isn't appropriate, though it seems reasonable without knowing exactly what's what here)